File: src/code/java/Q028.java
package code.java;

public class Q028 {
    public int strStr(String haystack, String needle) {
        int len1 = haystack.length(), len2 = needle.length();
        if (len1 < len2) {
            return -1;
        }
        if (len2 == 0) {
            return 0;
        }
        for (int i = 0; i <= len1 - len2; i++) {
            int j = 0;
            for (; j < len2; j++) {
                if (needle.charAt(j) != haystack.charAt(i + j)) {
                    break;
                }
            }
            if (j == len2) {
                return i;
            }
        }
        return -1;
    }
}

File: src/code/js/Q028.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function (haystack, needle) {
    var l1 = haystack.length, l2 = needle.length;
    var threshold = l1 - l2;
    if (l1 < l2) {
        return -1;
    }
    if (l2 === 0) {
        return 0;
    }
    for (var i = 0; i <= threshold; i++) {
        for (var j = 0; j < l2; j++) {
            if (haystack[i + j] !== needle[j]) {
                break;
            }
        }
        if (j === l2) {
            return i;
        }
    }
    return -1;
};
//怎么用KMP算法实现？